### PR TITLE
FreeBSD Testing / Release Automation

### LINKED ISSUE #1647: Cannot install poetry on freebsd using get-poetry.sh
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: FreeBSD 11.1-RELEASE-p1
- **Poetry version**: master

## Issue

Running `curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python` ends with following error on FreeBSD:

`pip install poetry` works fine.

```
Installing version: 0.12.17
Traceback (most recent call last):
  File "<stdin>", line 533, in _make_lib
  File "/usr/local/lib/python3.5/urllib/request.py", line 163, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/local/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/lib/python3.5/urllib/request.py", line 510, in error
    return self._call_chain(*args)
  File "/usr/local/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.5/urllib/request.py", line 590, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 959, in <module>
  File "<stdin>", line 955, in main
  File "<stdin>", line 356, in run
  File "<stdin>", line 489, in install
  File "<stdin>", line 509, in make_lib
  File "<stdin>", line 536, in _make_lib
RuntimeError: Could not find poetry-0.12.17-freebsd11.sha256sum file
```

### LINKED ISSUE #2159: Support for BSDs systems (FreeBSD)
I can't install on freebsd.

RuntimeError: could not find poetry-1.0.5-freebsd12.sha256sum file

### PR DESCRIPTION
**Pull request by neersighted**
Here's a first pass at FreeBSD CI/CD FreeBSD using Cirrus. There is no automated uploading yet, but such support could be added trivially -- the script [here](https://cirrus-ci.org/examples/#release-assets) is easily integrated.

Note that almost all behavior is modeled after the existing Actions. Also, we currently only run tests against FreeBSD 12 as that is the latest stable release. Additionally, we are using snapshots of the `STABLE` branch, as it represents a minimal delta that lets us test against upcoming `RELEASE` versions.

Some example build artifacts are attached to this PR.

This would resolve #2159 and resolve #1647.

[poetry-1.1.0a1-freebsd11.tar.gz](https://github.com/python-poetry/poetry/files/4466771/poetry-1.1.0a1-freebsd11.tar.gz)
[poetry-1.1.0a1-freebsd12.tar.gz](https://github.com/python-poetry/poetry/files/4466774/poetry-1.1.0a1-freebsd12.tar.gz)

### CODE REVIEW

**Comment by abn** [comment]
This is awesome! Thanks @neersighted! I think it might be great to have some form of CI in place as well, mainly because it would be good to be sure what we are shipping works :)

**Comment by neersighted** [comment]
I'm included to do that on a future PR -- but admittedly that may let it fall by the wayside. If you think it won't be too big for a single PR, I can see about adding FreeBSD CI as well. Note that as with uploading, additional integration work will be required by an Org member after merge.

**Comment by abn** [comment]
Feel free to add the add the CI changes to this one, happy to review that. Figured as much, regarding the uploading of the release tarball, we can tackle that once this is setup.

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - PYPACKAGE=$(printf '%s' $PYTHON | tr -d '.')
+    - SQLPACKAGE=$(printf '%s-sqlite3' $PYPACKAGE | sed 's/thon//')
+    - pkg install -y git-lite $PYPACKAGE $SQLPACKAGE
+  pip_script:
+    - $PYTHON -m ensurepip
+    - $PYTHON -m pip install pip -U
+    - $PYTHON -m pip install poetry -U --pre
+    - $PYTHON -m poetry config virtualenvs.in-project true
+  deps_script: $PYTHON -m poetry install
+  test_script: $PYTHON -m poetry run pytest -q --junitxml=junit.xml tests
```
Comment (abn) [inline .cirrus.yml:23]: I can see we are uploading the `junit.xml` below, but I am assuming this is for the check comments? Can we make sure this only comments on errors?

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    matrix:
+      - PYTHON: python2.7
+      - PYTHON: python3.5
+      - PYTHON: python3.6
+      - PYTHON: python3.7
+      - PYTHON: python3.8
+  python_script:
+    - PYPACKAGE=$(printf '%s' $PYTHON | tr -d '.')
+    - SQLPACKAGE=$(printf '%s-sqlite3' $PYPACKAGE | sed 's/thon//')
+    - pkg install -y git-lite $PYPACKAGE $SQLPACKAGE
```
Comment (abn) [inline .cirrus.yml:16]: Are these required?

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+freebsd_instance:
+  image_family: freebsd-12-1-snap
+
+test_task:
+  name: Tests (FreeBSD)
```
[the remark this answers is not in the document]
Comment (neersighted) [reply .cirrus.yml:5]: I can certainly make it a little more uniform -- I think `Tests / FreeBSD / PYTHON: python3.7` would be the closest we can get.

File: .cirrus.yml (hunk 1 above)
Comment (neersighted) [reply .cirrus.yml:23]: We can upload them only on failures if we don't want annotations for un-failed runs.

File: .cirrus.yml (hunk 2 above)
Comment (neersighted) [reply .cirrus.yml:16]: Not sure what you mean by are these required -- do we need `git` and `sqlite` bindings for the test suite to run? Yes, we do. Do we need to do string transforms to get the right package name? If we want pretty/readable test names without many environmental variables, yes we do. Take a look at the commit history to see what it looks like without the string manipulation.

Comment (abn) [reply .cirrus.yml:16]: Apologies. I should have been a bit more clear and probably more awake. I was curious about the sqllite package. Didn't occor this was being used by pytest. Ignore my comment.

**Pull request merged**